src/develop/imageop.h

```diff
@@ -378,6 +382,13 @@ typedef struct dt_iop_module_t
   void (*process_tiling)(struct dt_iop_module_t *self, struct dt_dev_pixelpipe_iop_t *piece, void *i, void *o,
                          const struct dt_iop_roi_t *roi_in, const struct dt_iop_roi_t *roi_out,
                          const int bpp);
+  /** NOTE: most likely, you do not want to implement those ^ two functions, but next two: */
+  /** the simplest variant of process(). you can only use OpeMP SIMD here, no intrinsics */
```

Comment:
Ope_n_MP

okay, so the logic is there is a stray process() if you want to, and a more restricted process_plain() and a rogue process_sse() where you can use inline assembly. but why keep process() and process_plain() ? sounds like making things overly complicated and inviting people to write both at the same time?